src/generator/operations.ts

```diff
@@ -199,17 +200,25 @@ export async function generateOperations(session: Session<CodeModel>): Promise<O
     for (const doc of values(paramDocs)) {
       clientText += `${doc}\n`;
     }
-    clientText += `func ${clientCtor}(${methodParams.join(', ')}) *${clientName} {\n`;
+    if (<boolean>session.model.language.go!.azureARM) {
+      clientText += `func ${clientCtor}(${methodParams.join(', ')}) (*${clientName}, error) {\n`;
+    } else {
+      clientText += `func ${clientCtor}(${methodParams.join(', ')}) *${clientName} {\n`;
```

Comment:
That's the idea. I'm not averse to returning `error` across the board though.

Reply:
I was thinking about that, in the off chance that we'll need to in the future.  However, presumably such a change wouldn't be breaking, assuming that we never export the generated client constructors.  If we think that we will want to export them, then making them return an error might be better (they aren't exported by default but can be enabled via a switch).

Reply:
I'll also add that I'm not really a fan of adding a vestigial return type "just in case", so am leaning towards not doing it.

Reply:
I like the idea of adding a switch. I also don't want to return a vestigial type but I expect we'll have to treat breaking changes to data plane code as breaking changes to the generator, if only because some other team will distribute generated code without a convenience layer, so it could help to allow the optional flexibility. Then again, I can also imagine a team who have painted themselves into a corner by distributing code generated without that option, so maybe it makes no difference. I think the right thing to do here depends on our DPG story and how we intend to support generated code.